youtube_dl/extractor/theplatform.py

```diff
@@ -144,11 +144,13 @@ def _real_extract(self, url):
         head = meta.find(_x('smil:head'))
         body = meta.find(_x('smil:body'))
 
-        f4m_node = body.find(_x('smil:seq//smil:video'))
-        if f4m_node is None:
-            f4m_node = body.find(_x('smil:seq/smil:video'))
-        if f4m_node is not None and '.f4m' in f4m_node.attrib['src']:
-            f4m_url = f4m_node.attrib['src']
+        node = body.find(_x('smil:seq//smil:video'))
+        if node is None:
+            node = body.find(_x('smil:seq/smil:video'))
+        if node is not None and '.m3u8' in node.attrib['src']:
+            formats = self._extract_m3u8_formats(node.attrib['src'], video_id)
```

Comment:
m3u8 can't be extracted now since `formats` will be overwritten by the code below.